### DISCUSSION TITLE
HttpClient behavior difference between .Net 6 and .Net 5: How to fix?

### DISCUSSION THREAD

**Discussion by raweibel** [dotnet/runtime#62046] [Q&A] [answered]
Hello everyone,
we are working on an .Net 6 application that is run on a corproate network which requires proxy authentication using the windows credentials. As we were running into problems using HttpClient, we wrote a small test application and noticed that when running in .Net 5 things work as expected, but in .Net 6 they do not.

**Our test code:**
`var handler = new HttpClientHandler();`
`handler.DefaultProxyCredentials = CredentialCache.DefaultCredentials;`
`var client = new HttpClient(handler);`
`var response = await client.GetAsync(requestedURL);`
`Console.WriteLine($"Response: {response.StatusCode}");`

**Expected result:** 
_Response: Ok_

**Result in .Net 5:**
_Response: Ok_

**Result in .Net 6:**
_Unhandled exception. System.ComponentModel.Win32Exception (0x80090308): The token supplied to the function is invalid
   at System.Net.NTAuthentication.GetOutgoingBlob(Byte[] incomingBlob, Boolean throwOnError, SecurityStatusPal& statusCode)_

Does anyone know if this is a known problem? Is there a way to fix this in .Net 6?

I'm happy to provide more detail if needed.

**Comment by huoyaoyuan** [comment, marked as answer]
You could open an issue with more details. The exception message looks related to your credentials.
